crypto/signer.go

```diff
@@ -147,3 +147,29 @@ func (s *StaticSigner) Sign(_ []byte) ([]byte, error) {
 func (s *StaticSigner) Public() PublicKey {
 	return s.pubKey
 }
+
+// SignerV2 allows to sign authorized data.
+type SignerV2 interface {
+	SignData(data []byte) (Signature, error)
+}
+
+type mergedSigners struct {
+	Signer
+	SignerV2
+}
+
+// MergeSigners merges [Signer] and [SignerV2] implementations to provide
+// [Signer] for functions trying to switch to [SignerV2] usage.
+func MergeSigners(s Signer, s2 SignerV2) Signer {
```

Comment:
do you have a practical caller for this? imo, too much helper code for temporary cases. do not insist

Reply:
i changed my mind about adding it, but I fixed the wrong commit. No more `MergeSigners`, only `OverlapSigner`, which is most likely required for now